### DISCUSSION TITLE
Copying a chart to the clipboard?

### DISCUSSION THREAD

**Discussion by tonyvsuk** [utelle/wxchartdir#9] [Q&A] [answered]
Hello,

I'm trying to get a bitmap of my chart onto the clipboard, but struggling to do so (I keep getting an invalid bitmap message).

Has anyone else succeeded?

My code is below. I've tried two different ways to get at the data (pBMPData below), neither worked for me.

Thanks in advance,

Tony.

	if (wxTheClipboard->Open())
	{
		BaseChart *pChart = getChart();

		if (pChart != NULL)
		{
			MemBlock block = pChart->makeChart(Chart::BMP);

			//size_t buflen = block.len - sizeof(BITMAPFILEHEADER);

			const char *pBMPData = block.data;
			//const char *pBMPData = (block.data + sizeof(BITMAPFILEHEADER));

			wxBitmap bmp(pBMPData);
			wxBitmapDataObject *pBMPObject = new wxBitmapDataObject(bmp);

			wxTheClipboard->SetData(pBMPObject);

			wxTheClipboard->Close();
		}
	}

**Comment by utelle** [comment, marked as answer]
Unfortunately, the memory block returned by `makeChart(Chart::BMP)` can't be passed directly to a `wxBitmap` constructor. First, you have to convert the memory block to a `wxImage` object that then can be passed to a `wxBitmap` constructor. 

The code looks like this:
```C++
  MemBlock block = c->makeChart(Chart::BMP);
  wxMemoryInputStream in(block.data, block.len);
  wxBitmap bmp(wxImage(in, wxBITMAP_TYPE_BMP));
```
